src/runtime/stream.rs

```diff
@@ -74,9 +74,13 @@ impl AsyncTcpStream {
             timeout(connect_timeout, stream_future).await??
         };
 
-        stream.set_keepalive(Some(KEEPALIVE_TIME))?;
         stream.set_nodelay(true)?;
 
+        let socket = socket2::Socket::from(stream.into_std()?);
```

Comment:
Tokio 1.0 currently doesn't expose any way to set keepalive on `TcpSocket`, but as a workaround, we can convert it to a `socket2` socket, set keepalive on that, and then convert it back.